test/integration/filters/response_metadata_filter.cc

```diff
@@ -16,42 +16,39 @@ class ResponseMetadataStreamFilter : public Http::PassThroughFilter {
 public:
   // Inserts one new metadata_map.
   Http::FilterHeadersStatus encodeHeaders(Http::HeaderMap&, bool) override {
-    Http::MetadataMap metadata_map = {
-        {"headers", "headers"}, {"duplicate", "duplicate"}, {"remove", "remove"}};
+    Http::MetadataMap metadata_map = {{"headers", "headers"}, {"duplicate", "duplicate"}};
     Http::MetadataMapPtr metadata_map_ptr = std::make_unique<Http::MetadataMap>(metadata_map);
-    decoder_callbacks_->encodeMetadata(std::move(metadata_map_ptr));
```

Comment:
Are there HCM unit tests we should update as well?

Reply:
I think adding new metadata was only covered in integration test.